hub/core/chunk_engine.py

```diff
@@ -822,6 +879,185 @@ def update(
+                else None
+            )
+            if decompress is False:
+                samples_to_move = [
+                    Sample(buffer=sample_bytes, shape=new_shape, compression=compression)  # type: ignore
+                ] + samples_to_move
+            else:
+                samples_to_move = [
+                    Sample(array=sample_bytes, shape=new_shape, compression=compression)  # type: ignore
+                ] + samples_to_move
+
+        samples_to_move.reverse()
+
+        return samples_to_move
+
+    def __rechunk(self, chunk, chunk_row):
+        samples_to_move = self.__get_samples_to_move(chunk=chunk)
+        num_samples = len(samples_to_move)
+        if num_samples == 0:
+            return
+        new_chunk = self._create_new_chunk(register=True, row=chunk_row)
+
+        self.chunk_id_encoder.decrease_samples(row=chunk_row, num_samples=num_samples)
+        self.chunk_id_encoder.decrease_samples(
+            row=chunk_row + 1, num_samples=num_samples
+        )
+        chunk.pop_multiple(num_samples=len(samples_to_move))
+
+        samples, _ = self._sanitize_samples(samples_to_move)
+        self._samples_to_chunks(
+            samples,
+            start_chunk=new_chunk,
+            register=True,
+            update_commit_diff=True,
+            extend=False,
+            fit_row=chunk_row + 1,
+        )
+
+    def __try_merge_with_next_chunk(self, chunk, row) -> bool:
+        next_chunk_id = self.chunk_id_encoder.get_next_chunk_id(row)
+        if next_chunk_id is None:
+            return False
+        next_chunk_row = row + 1
+        next_chunk_name = ChunkIdEncoder.name_from_id(next_chunk_id)  # type: ignore
+        next_chunk_commit_id = self.get_chunk_commit(next_chunk_name)
+        chunk_key = get_chunk_key(self.key, next_chunk_name, next_chunk_commit_id)
+        next_chunk_size = self.cache.get_object_size(chunk_key)
+        next_chunk = self.get_chunk_from_chunk_id(int(next_chunk_id))
+
+        if next_chunk_size + chunk.num_data_bytes < next_chunk.min_chunk_size:
+            # merge with next chunk
+            samples_to_move = self.__get_chunk_samples(chunk=next_chunk)
+            num_samples = len(samples_to_move)
+            if num_samples == 0:
+                return True
+
+            next_chunk.pop_multiple(num_samples=num_samples)
+            samples, _ = self._sanitize_samples(samples_to_move)
+            self._samples_to_chunks(
+                samples,
+                start_chunk=chunk,
+                register=True,
+                update_commit_diff=True,
+                append_to_end=True,
+                extend=False,
+                fit_row=row,
+            )
+            self.chunk_id_encoder.delete_chunk_id(row=next_chunk_row)
+            del self.cache[next_chunk.key]  # type: ignore
+            return True
+
+        return False
+
+    def __try_merge_with_previous_chunk(self, chunk, row) -> bool:
```

Comment:
both this and try merge with next chunk are practically identical, needs refactor